### PR TITLE
move hook tests to adapter zone

### PR DESCRIPTION
**Pull request by colin-rogers-dbt**
resolves #

### Checklist

- [ ] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Review by nssalian — COMMENTED** [review]
Could we move the .sql files into a python file and import them into the tests? Possibly combine the 2 SQL inputs into one `seeds.py` and import where needed.

**Comment by colin-rogers-dbt** [comment]
> Could we move the .sql files into a python file and import them into the tests? Possibly combine the 2 SQL inputs into one `seeds.py` and import where needed.

Meant to open this as a draft, still iterating!

**Review by McKnight-42 — APPROVED** [review]

File: tests/adapter/dbt/tests/adapter/hooks/data/seed_run.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+drop table if exists {schema}.on_run_hook;
+
+create table {schema}.on_run_hook (
```
Comment (mikealfare) [inline tests/adapter/dbt/tests/adapter/hooks/data/seed_run.sql:4]: It's minor, but if you always want `on_run_hook` to look like `on_model_hook`, you could just create this table from the other:
```
drop table if exists {schema}.on_run_hook;
create table {schema}.on_run_hook as
select * from {schema}.on_model_hook
where 0 = 1
```

File: tests/adapter/dbt/tests/adapter/hooks/fixtures.py [hunk 2]
```diff
@@ -30,18 +30,17 @@
...
    insert into {{ target.schema }}.on_run_hook (
-        "state",
-        "target.dbname",
-        "target.host",
-        "target.name",
-        "target.schema",
-        "target.type",
-        "target.user",
-        "target.pass",
-        "target.port",
```
Comment (mikealfare) [inline tests/adapter/dbt/tests/adapter/hooks/fixtures.py:41]: I want to confirm, we're specifically getting rid of `port`, right?

File: tests/adapter/dbt/tests/adapter/hooks/test_model_hooks.py [hunk 3]
```diff
@@ -91,22 +87,23 @@ def setUp(self, project):
...
+            "target_user",
+            "target_pass",
             "run_started_at",
             "invocation_id",
         ]
         field_list = ", ".join(['"{}"'.format(f) for f in fields])
-        query = "select {field_list} from {schema}.on_model_hook where state = '{state}'".format(
-            field_list=field_list, schema=project.test_schema, state=state
+        query = (
+            "select {field_list} from {schema}.on_model_hook where test_state = '{state}'".format(
```
Comment (mikealfare) [inline tests/adapter/dbt/tests/adapter/hooks/test_model_hooks.py:104]: Go go gadget f-string!

File: tests/adapter/dbt/tests/adapter/hooks/test_model_hooks.py [hunk 4]
```diff
@@ -165,9 +161,12 @@ def models(self):
 
     def test_pre_and_post_run_hooks(self, project, dbt_profile_target):
         run_dbt()
-
-        self.check_hooks("start", project, dbt_profile_target["host"])
-        self.check_hooks("end", project, dbt_profile_target["host"])
+        if "host" in dbt_profile_target:
```
Comment (mikealfare) [inline tests/adapter/dbt/tests/adapter/hooks/test_model_hooks.py:164]:
```
self.check_hooks("start", project, dbt_profile_target.get("host"))
self.check_hooks("end", project, dbt_profile_target.get("host"))
```

File: tests/adapter/dbt/tests/adapter/hooks/test_model_hooks.py [hunk 5]
```diff
@@ -229,8 +227,12 @@ def models(self):
     def test_pre_post_model_hooks_refed(self, project, dbt_profile_target):
         run_dbt()
 
-        self.check_hooks("start", project, dbt_profile_target["host"], count=1)
-        self.check_hooks("end", project, dbt_profile_target["host"], count=1)
+        if "host" in dbt_profile_target:
```
Comment (mikealfare) [inline tests/adapter/dbt/tests/adapter/hooks/test_model_hooks.py:230]: Same as line 164

**Pull request merged**